### PR TITLE
Fix for issue 6602 - warning when running composer as another user

### PR DESCRIPTION
**Pull request by wsandin**
Raising a warning if the current user doesn't match the initial ownership of the $home directory. 
Not sure how this works on let's say Windows, might have to add some extra checks for that.

Would love some feedback on how I can improve this PR in order to merge into master.

### CODE REVIEW

File: src/Composer/Command/SelfUpdateCommand.php [hunk 1]
```diff
@@ -93,7 +93,9 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $cacheDir = $config->get('cache-dir');
         $rollbackDir = $config->get('data-dir');
         $home = $config->get('home');
+        $homeOwner = posix_getpwuid(fileowner($home));
         $localFilename = realpath($_SERVER['argv'][0]) ?: $_SERVER['argv'][0];
+        $composeUser = posix_getpwuid(posix_geteuid());
```
Comment (Seldaek) [inline src/Composer/Command/SelfUpdateCommand.php]: This check should only be done if posix_getpwuid / posix_geteuid are defined and callable, as the posix functions are typically disabled or not available in some environments and we don't want to fail the update in those cases.

Comment (wsandin) [reply src/Composer/Command/SelfUpdateCommand.php]: Thanks @Seldaek. Will have a look into how I can improve this.

**Comment by Seldaek** [comment]
Thanks

**Comment by wsandin** [comment]
Thanks, would love to contribute more to composer in the future.

**Pull request merged**